PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/layers/data_link_layer.cpp

```diff
@@ -0,0 +1,93 @@
+#include "layers/data_link_layer.h"
+
+/**
+ * @brief Updates the shared credit limit values based on the received DLLP information
+ * @param flit Flit containing the DLLPs
+ * @param P_SHARED_CREDIT_LIMIT Array of shared credit limit for posted traffic
+ * @param NP_SHARED_CREDIT_LIMIT Array of shared credit limit for non-posted traffic
+ * @param CPL_SHARED_CREDIT_LIMIT Array of shared credit limit for completion traffic
+ * @param FI1 First indication (FI1) flag for credit update
+ * @param FI2 Second indication (FI2) flag for credit update
+*/
+void DataLinkLayer::updateCreditLimit(Flit flit, int& P_SHARED_CREDIT_LIMIT[2], int& NP_SHARED_CREDIT_LIMIT[2], int& CPL_SHARED_CREDIT_LIMIT[2], bool& FI1, bool& FI2) {
+	// Static variables to store the flags and state across function calls
+	static bool discard, pFc1Flag, npFc1Flag, cplFc1Flag, pFc2Flag, npFc2Flag, cplFc2Flag;
```

Comment:
You're here only checking for 3 flags and not 6. The 6 flags should be as follows:
P Shared
NP Shared
Cpl Shared
P Dedicated
NP Dedicated
Cpl Dedicated

N.B.: Dedicated credits are always 0 in our use case